modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java

```diff
@@ -389,17 +401,23 @@ protected void beforeCreateDocument(DocumentModel doc) {
     @Override
     public void start(ComponentContext context) {
         super.start(context);
-        plugins.clear();
-        List<PluginDescriptor> descriptors = getDescriptors(XP_PLUGINS);
-        for (PluginDescriptor descriptor : descriptors) {
+        fillRegistry(XP_PLUGINS, plugins, PluginDescriptor.class);
+        fillRegistry(XP_EXPORTS, exporters, ExporterDescriptor.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected <T> void fillRegistry(String xp, Map<String, T> registry, Class<?> descriptorClass) {
+        registry.clear();
+        List<Descriptor> descriptors = getDescriptors(xp);
+        for (Descriptor descriptor : descriptors) {
             try {
                 Class<?> clazz = Class.forName(descriptor.getKlass());
-                Constructor<?> constructor = clazz.getConstructor(PluginDescriptor.class);
-                Plugin<?> plugin = (Plugin<?>) constructor.newInstance(descriptor);
-                plugins.put(descriptor.getId(), plugin);
+                Constructor<?> constructor = clazz.getConstructor(descriptorClass);
+                T instance = (T) constructor.newInstance(descriptor);
```

Comment:
You can probably avoid the cast by declaring in the signature `Class<T> descriptorClass` then doing `Constructor<T> constructor = ...`

Reply:
As discussed, will need to cast anway given the Class.forName signature returning Class<?>